src/components/menu/menu.types.ts

```diff
@@ -2,7 +2,7 @@ import { ListItem } from '../list/list-item.types';
 
 export type OpenDirection = 'left' | 'right';
 
-export interface MenuItem extends ListItem {
+export interface MenuItem<T = any> extends ListItem<T> {
```

Comment:
Just curious here:

Huh… so, with the `export interface MenuItem extends ListItem`, `MenuItem` doesn't automatically inherit the `<T = any>` part from `ListItem`? Seems like it should, logically, so (assuming it's not inherited) is that just a flaw in Typescript, and this is basically a workaround, or does it actually make sense that `export interface MenuItem extends ListItem` and `export interface MenuItem<T = any> extends ListItem<T>` should mean different things and this is a feature of Typescript, rather than a flaw? 🙂

Reply:
I think it's by design, otherwise you would not be able to do stuff like this
```ts
export interface HorseItems extends ListItem<Horse> {}
export interface CowItems extends ListItem<Cow> {}
```

Maybe not the best example here, but I hope you get the idea :sweat_smile:

Reply:
Ah! Yeah, of course! That makes sense!

(I think your example was very clear and helpful! 🙏)